Index out of range error on azurerm_batch_pool

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.6

### AzureRM Provider Version

3.41.0

### Affected Resource(s)/Data Source(s)

azurerm_batch_pool

### Terraform Configuration Files

```hcl
data "azurerm_storage_account" "platform_storage" {
    name = "somestorageaccount"
    resource_group_name = var.azure_resource
}

resource "azurerm_batch_account" "platform_batch_account" {
    name = "somebatchaccount"
    resource_group_name = var.azure_resource
    location = var.azure_region
    pool_allocation_mode = "BatchService"
    storage_account_id = data.azurerm_storage_account.platform_storage.id
    storage_account_authentication_mode = "StorageKeys"
}

resource "azurerm_batch_pool" "platform_batch_pool" {
    name = "somebatchpool"
    resource_group_name = var.azure_resource
    account_name = azurerm_batch_account.platform_batch_account.name
    node_agent_sku_id = "batch.node.ubuntu 20.04"
    vm_size = "Standard A1"

    auto_scale {
        evaluation_interval = "PT15M"

        formula = <<EOF
        startingNumberOfVMs = 1;
        maxNumberofVMs = 25;
        pendingTaskSamplePercent = $PendingTasks.GetSamplePercent(180 * TimeInterval_Second);
        pendingTaskSamples = pendingTaskSamplePercent < 70 ? startingNumberOfVMs : avg($PendingTasks.GetSample(180 *   TimeInterval_Second));
        $TargetDedicatedNodes=min(maxNumberofVMs, pendingTaskSamples);
        EOF
    }

    storage_image_reference {
        publisher = "microsoft-azure-batch"
        offer     = "ubuntu-server-container"
        sku       = "20-04-lts"
        version   = "latest"
    }

    container_configuration {
        type = "DockerCompatible"
        container_registries {
            registry_server = "pathtoregistry"
            user_name = "user"
            password = data.azurerm_key_vault_secret.docker_key.value
        }
    }

    start_task {
        command_line       = "echo 'hello world'"
        task_retry_maximum = 1
        wait_for_success   = false

        common_environment_properties = {
            ...redacted
        }

        container {
            image_name  = "container_image_name"
        }

        user_identity {
            auto_user {
                elevation_level = "Admin"
                scope           = "Task"
            }
        }
    }
}

resource "azurerm_batch_job" "platform_batch_job" {
    name = "somebatchjob"
    batch_pool_id = azurerm_batch_pool.platform_batch_pool.id
}
```


### Debug Output/Panic Output

```shell
module.azure.azurerm_batch_pool.platform_batch_pool: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with module.azure.azurerm_batch_pool.platform_batch_pool,
│   on azure/ingest_rollup.tf line 15, in resource "azurerm_batch_pool" "platform_batch_pool":
│   15: resource "azurerm_batch_pool" "platform_batch_pool" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-azurerm_v3.41.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 79 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/batch.ExpandBatchPoolStartTask({0xc001c5eff0?, 0x73d019b?, 0xc001df88e8?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/batch/batch_pool.go:713 +0xb57
github.com/hashicorp/terraform-provider-azurerm/internal/services/batch.resourceBatchPoolCreate(0xc0000ebf00, {0x6903460?, 0xc001770400})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/batch/batch_pool_resource.go:868 +0xa7c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x7b97560?, {0x7b97560?, 0xc001cc10e0?}, 0xd?, {0x6903460?, 0xc001770400?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000afc2a0, {0x7b97560, 0xc001cc10e0}, 0xc001cfcf70, 0xc0000ebd80, {0x6903460, 0xc001770400})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002d1a28, {0x7b97560?, 0xc001cc0fc0?}, 0xc001a42410)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001160820, {0x7b97560?, 0xc001cc0780?}, 0xc0005bc000)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x6fdc140?, 0xc001160820}, {0x7b97560, 0xc001cc0780}, 0xc002615f80, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001ca1e0, {0x7ba7900, 0xc00059c680}, 0xc001cdfe60, 0xc0013bee70, 0xbe56da0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0001ca1e0, {0x7ba7900, 0xc00059c680}, 0xc001cdfe60, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.41.0_x5 plugin crashed!
```


### Expected Behaviour

Terraform should have created the Azure Batch Pool

### Actual Behaviour

It crashed

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/batch/batch_pool.go
+++ b/internal/services/batch/batch_pool.go
@@ -710,8 +710,8 @@ func ExpandBatchPoolStartTask(list []interface{}) (*batch.StartTask, error) {
 			if containerRunOptions, ok := settingMap["run_options"]; ok {
 				containerSettings.ContainerRunOptions = utils.String(containerRunOptions.(string))
 			}
-			if settingMap["registry"].([]interface{})[0] != nil {
-				containerRegMap := settingMap["registry"].([]interface{})[0].(map[string]interface{})
+			if registries, ok := settingMap["registry"].([]interface{}); ok && len(registries) > 0 && registries[0] != nil {
+				containerRegMap := registries[0].(map[string]interface{})
 				if containerRegistryRef, err := expandBatchPoolContainerRegistry(containerRegMap); err == nil {
 					containerSettings.Registry = containerRegistryRef
 				}

